src/scenes/creator/final/name.tsx

```diff
@@ -1,28 +1,93 @@
-import React from "react";
+import React, { useEffect } from "react";
 import styles from "./name.module.scss";
 import { Label } from "../../../components/label";
 import { useDispatch, useSelector } from "react-redux";
-import { nameSelector } from "../../../store/name/selectors";
-import { updateName } from "../../../store/name/nameSlice";
+import { updateDefaultName, updateName } from "../../../store/name/nameSlice";
+import {
+  addCheeseSelector,
+  cheesesSelector,
+} from "../../../store/cheese/selectors";
+import { addMeatSelector, meatsSelector } from "../../../store/meat/selectors";
+import {
+  addDressingSelector,
+  dressingsSelector,
+} from "../../../store/dressing/selectors";
+import { vegetablesSelector } from "../../../store/vegetables/selectors";
+import { addEggSelector, eggsSelector } from "../../../store/egg/selectors";
+import { spreadSelector } from "../../../store/spreads/selectors";
+import { toppingSelector } from "../../../store/topping/selectors";
 
 export function Name() {
   const dispatch = useDispatch();
 
-  const paniniName = useSelector(nameSelector);
+  const addCheese = useSelector(addCheeseSelector);
+  const currentCheese = useSelector(cheesesSelector);
+  const cheese = addCheese ? currentCheese : [];
+
+  const addMeat = useSelector(addMeatSelector);
+  const currentMeat = useSelector(meatsSelector);
```

Comment:
`const meat = addMeat ? currentMeat : [];`

here, and with other ingredients you can try to move this logic to a seperate selector. 
it's good to keep "the buisness logic" in one place, so in this case in redux related files. 
You can combine selectors addMeatSelector and meatsSelector 
 
 https://redux.js.org/usage/deriving-data-selectors#createselector-overview

Reply:
I could finally use createSelector function :) It's done!